### PR TITLE
Zadanie_3_Kostiantyn_Ivanov_Neuron

### PR DESCRIPTION
**Pull request by Kostya-Iva**

### CODE REVIEW

**Comment by farqlia** [comment]
- sprytnie z pytorcha dane pobrane 
- super, że baseline jest taki najprostszy i ma tylko jedną warstwę ukrytą a nie np. dwie czy więcej
- "w przypadku zbioru danych "MNIST" dane są raczej "zbalansowane": można było tak czy siak wyświetlić liczbę labeli na klasę, tak dla porządku 
- wyświetlanie wyników dla zbioru testowego może podświadomie nakierować na pewne obszary badań (mamy więc wyciek wiedzy), więc dla czystości metodologii lepiej testowy na samym końcu używać, już po wybraniu modelu
- można było wyświetlić na jednym wykresie krzywe dla CNN i MLP (np. tylko dla jednego zbioru), to też jest przydatne porównanie
- szkoda, że nie dodałeś oddzielnie BatchNorm i Dropout, bo wtedy byłby widoczny efekt dodania tylko jednego elementu do sieci, a tak to nie wiemy, który faktycznie spowodował większą poprawę. Ale uznam, że to wina zasobów. 
- zamiast powtarzać pętlę do trenowania dla każdego podzbioru, to można było ją ubrać do funkcji

Oprócz tego zbioru testowego to myślę, że jest poprawnie.

**Review by Tomoo3 — COMMENTED** [review]
Zadanie dobrze wykonane, obszerne opisy i wizualizacje, wszystko elegancko czytelne, nie ma do czego się przyczepić

**Review by fidok15 — COMMENTED** [review]
- zbiór testowy jest używany od razu, powinien być użyty na samym końcu do sprawdzenia finalnego modelu, a dostosowywanie praramtrów powino się odbyć z zbiorem validacyjnym i treningowym
- bogate opisy po każdej sekcji 
- dobrz, że jest sprawdzenie dwóch optimizerów 
- brakuje mi sprawdzenia zachowania modelu, dla więceji niż 5 epok 
- brak pokazania, z którymi przypadkami model sobie nie radzi 
- dobrze wyciągnięte wnioski
